include/deal.II/numerics/fe_field_function.templates.h

```diff
@@ -140,7 +141,14 @@ namespace Functions
     FEValues<dim> fe_v(mapping, cell->get_fe(), quad,
                        update_gradients);
     fe_v.reinit(cell);
-    fe_v.get_function_gradients(data_vector, gradients);
+
+    // Unfortunately we still need a temporary argument as we want to
+    // evaluate a gradient of a (generally) multicomponent function at
+    // a single quadrature point.
+    std::vector< std::vector<Tensor<1,dim,number> > > vgrads
+    (1,  std::vector<Tensor<1,dim,number> >(n_components) );
+    fe_v.get_function_gradients(data_vector, vgrads);
+    gradients = std::vector<Tensor<1,dim,number> >(vgrads[0].begin(), vgrads[0].end());
```

Comment:
> So, to wrap up the discussion about the validity of the patch -- do I understand it correctly that you here are simply restoring the previous code, and that what confuses me is that the comment was wrong in that it claimed that the temp variable was necessary because of the data type, when in reality it was necessary because of the number of vector arguments?

exactly.